src/osd/PeeringState.cc

```diff
@@ -2651,7 +2650,8 @@ void PeeringState::activate(
 
   if (is_primary()) {
     // only update primary last_epoch_started if we will go active
-    if (acting.size() >= pool.info.min_size) {
+    if ((acting.size() >= pool.info.min_size) &&
+	 pool.info.stretch_set_can_peer(acting, get_osdmap().get(), NULL)) {
```

Comment:
The repeated acting.size() >= pool.info.min_size was already not great.  With the added stretch_set_can_peer condition, I'd like some kind of helper, maybe acting_set_writeable?

Reply:
I thought about this but it's still more lines of code to do a helper function than to condense down the three places we check.

Reply:
Lines of code is not the concern.  The concern is that we'll fail to update one of these if we have to modify a condition.  I really do want a helper here.

Reply:
Fair enough.